### PR TITLE
343 broadcasts by area

### PR DESCRIPTION
**Pull request by domangi**
[WIP]

### CODE REVIEW

File: backend/app/models/statistic/broadcast.rb [hunk 1]
```diff
@@ -81,6 +81,18 @@ def self.global_average_per_impression(impressions_table)
       )
     end
 
+    def approval_by_state_codes
+      results = {}
+      User.all.distinct.pluck(:state_code).each do |state_code|
```
[the remark this answers is not in the document]
Comment (domangi) [reply backend/app/models/statistic/broadcast.rb]: I think the result is the same, but using distinct in front of pluck, results in making the uniqueness operation on the database which should be more efficient:

```ruby
User.all.distinct.pluck(:state_code)
# => SELECT DISTINCT "users"."state_code" FROM "users"
```

File: backend/app/models/statistic/broadcast.rb [hunk 2]
```diff
@@ -81,6 +81,18 @@ def self.global_average_per_impression(impressions_table)
...
+        results[state_code] = approval_by_state(state_code)
```
[the remark this answers is not in the document]
Comment (domangi) [reply backend/app/models/statistic/broadcast.rb]: Yes. A single query that computes the approvals grouped by each state_code.

[the remark this answers is not in the document]
Comment (domangi) [reply backend/app/models/statistic/broadcast.rb]: Should be fixed now through a database group_by and select avg(...)

File: backend/app/models/statistic/broadcast.rb (hunk 1 above)
[the remark this answers is not in the document]
Comment (roschaefer) [reply backend/app/models/statistic/broadcast.rb]: I didn't know that you can put `distinct` in front of `pluck`, learned sth. new! :+1:

**Pull request merged**